ts/src/coinmetro.ts

```diff
@@ -0,0 +1,1919 @@
+            'Deposit': 'transaction',
+            'Withdraw': 'transaction',
+            'Order': 'trade',
+        };
+        return this.safeString (types, type, type);
+    }
+
+    async createOrder (symbol: string, type: OrderType, side: OrderSide, amount, price = undefined, params = {}) {
+        /**
+         * @method
+         * @name coinmetro#createOrder
+         * @description create a trade order
+         * @see https://documenter.getpostman.com/view/3653795/SVfWN6KS#a4895a1d-3f50-40ae-8231-6962ef06c771
+         * @param {string} symbol unified symbol of the market to create an order in
+         * @param {string} type 'market' or 'limit'
+         * @param {string} side 'buy' or 'sell'
+         * @param {float} amount how much of currency you want to trade in units of base currency
+         * @param {float} [price] the price at which the order is to be fullfilled, in units of the quote currency, ignored in market orders
+         * @param {object} [params] extra parameters specific to the exchange API endpoint
+         * @param {float} [params.cost] the quote quantity that can be used as an alternative for the amount in market orders
+         * @param {string} [params.timeInForce] "GTC", "IOC", "FOK", "GTD"
+         * @param {number} [params.expirationTime] timestamp in millisecond, for GTD orders only
+         * @param {float} [params.triggerPrice] the price at which a trigger order is triggered at
+         * @param {float} [params.stopLossPrice] *margin only* The price at which a stop loss order is triggered at
+         * @param {float} [params.takeProfitPrice] *margin only* The price at which a take profit order is triggered at
+         * @param {bool} [params.margin] true for creating a margin order
+         * @param {string} [params.fillStyle] fill style of the limit order: "sell" fulfills selling quantity "buy" fulfills buying quantity "base" fulfills base currency quantity "quote" fulfills quote currency quantity
+         * @param {string} [params.clientOrderId] client's comment
+         * @returns {object} an [order structure]{@link https://docs.ccxt.com/#/?id=order-structure}
+         */
+        await this.loadMarkets ();
+        const market = this.market (symbol);
+        const request = {
+            'orderType': type,
+        };
+        let precisedAmount = undefined;
+        if (amount !== undefined) {
+            precisedAmount = this.amountToPrecision (symbol, amount);
+        }
+        let cost = this.safeNumber (params, 'cost');
+        params = this.omit (params, 'cost');
+        if (type === 'limit') {
+            if ((price === undefined) && (cost === undefined)) {
+                throw new ArgumentsRequired (this.id + ' createOrder() requires a price or params.cost argument for a ' + type + ' order');
+            } else if ((price !== undefined) && (amount !== undefined)) {
+                const costString = Precise.stringMul (price.toString (), precisedAmount.toString ());
```

Comment:
also, we should never do `numer.toString()`, we should we `this.numberToString()` instead

Reply:
Affirmative